provider/internal/image.go

```diff
@@ -599,7 +603,9 @@ func (ia *ImageArgs) validate(supportsMultipleExports, preview bool) (controller
 			multierr = errors.Join(multierr, newCheckFailure(err, "cacheFrom[%d]", idx))
 			continue
 		}
-		cacheFrom = append(cacheFrom, cache)
+		if cache != nil {
+			cacheFrom = append(cacheFrom, cache)
+		}
```

Comment:
The gist of the problem is that `validate` may or may not return a parsed cache entry; for gha cache entries in a non-gha environment, it may prefer to ignore the entry. Appending a `nil` entry here would otherwise cause a panic later.